.chronus/changes/json-converter-2024-6-24-16-48-41.md

```diff
@@ -0,0 +1,7 @@
+---
+changeKind: feature
+packages:
+  - "@azure-tools/typespec-client-generator-core"
+---
+
+Add `@hasJsonConverter` for csharp only to indicate if JSON converter is needed
```

Comment:
lets rename to @useSystemTextJsonConverter to make it more explicit what this is used for.

Reply:
renamed

Reply:
I'll say it feels quite weird having this at the root of Tcgc namespace with this name that is very csharp specific

Reply:
I think this is indeed csharp specific only decorator.
@timotheeguerin Are you implying we should keep it within csharp sub-namespace if we ensure it will not be used for other languages?

Reply:
Yeah just feels like this is not super compatible with the new approach of using a shared name and scope

Reply:
We have decided that language-specific decorators will go in the main namespace because of grow-up, but they will have to specify the scope, and we have an [issue](https://github.com/Azure/typespec-azure/issues/1334) to have linter warnings if a decorator is applied to emitters that won't do anything with it.

My question with this PR is more: I thought we had decided that there was a use case for Java here too, and that there was going to be a redesign of this decorator (most likely just a rename), that would make it compatible with Java's scenario as well